src/java/org/apache/cassandra/service/accord/serializers/WaitingOnSerializer.java

```diff
@@ -73,22 +68,17 @@ public static ByteBuffer serialize(TxnId txnId, WaitingOn waitingOn) throws IOEx
         if (waitingOn.appliedOrInvalidated != null)
             appliedOrInvalidatedLength = (txnIdCount + 63) / 64;
 
-        Timestamp executeAtLeast = waitingOn.executeAtLeast();
-        ByteBuffer out = ByteBuffer.allocate(1 + (executeAtLeast == null ? 0 : CommandSerializers.timestamp.serializedSize())
-                                             + TypeSizes.sizeofUnsignedVInt(keyCount) + TypeSizes.sizeofUnsignedVInt(txnIdCount)
+        ByteBuffer out = ByteBuffer.allocate(TypeSizes.sizeofUnsignedVInt(waitingOnLength)
+                                             + (waitingOn.appliedOrInvalidated == null ? 0 : TypeSizes.sizeof(waitingOnLength - appliedOrInvalidatedLength))
```

Comment:
I think we need to have `TypeSizes.sizeofUnsignedVInt` here instead of just `sizeOf`